src/main/java/com/onelogin/sdk/conn/Client.java

```diff
@@ -2438,6 +2442,70 @@ public SAMLEndpointResponse getSAMLAssertionVerifying(String appId, String devid
 	}
 
+	///////////////////////
+	//  SMART MFA        //
+	///////////////////////
+
+	/**
+	 * Validates a User
+     *
+	 * @param validateUserParams Smart MFA User data (user_identifier, email, phone, context, risk_threshold,
+     *                                                 firstname, lastname, expires_in)
+	 *
+	 * @return SmartMFA
+	 *
+	 * @throws OAuthSystemException - if there is a IOException reading parameters of the httpURLConnection
+	 * @throws OAuthProblemException - if there are errors validating the OneloginOAuthJSONResourceResponse and throwOAuthProblemException is enabled
+	 * @throws URISyntaxException - if there is an error when generating the target URL at the URIBuilder constructor
+	 * @throws ErrorResourceInitialization
+	 *
+	 * @see com.onelogin.sdk.model.SmartMFA
+	 * @see <a target="_blank" href="https://developers.onelogin.com/api-docs/2/smart-mfa/validate-user">Validate a Use documentation</a>
+	 */
+	public SmartMFA validateUser(Map<String, Object> validateUserParams) throws OAuthSystemException, OAuthProblemException, URISyntaxException, ErrorResourceInitialization {
+			int versionId = settings.getVersionId("SMART_MFA_VALIDATE_USER");
+			URIBuilder url = new URIBuilder(settings.getURL(Constants.SMART_MFA_VALIDATE_USER, versionId));
+
+			return (SmartMFA) createResource(SmartMFA.class, validateUserParams, url, versionId);
+	}
+
+	/**
+	 * Verify MFA Token
+     *
+	 * @param stateToken The state_token value returned from the Validate a User endpoint
+	 * @param OTPToken The MFA token that was sent to the user via Email or SMS using the Validate a User endpoint
+	 *
+	 * @return boolean
+	 *
+	 * @throws OAuthSystemException - if there is a IOException reading parameters of the httpURLConnection
+	 * @throws OAuthProblemException - if there are errors validating the OneloginOAuthJSONResourceResponse and throwOAuthProblemException is enabled
+	 * @throws URISyntaxException - if there is an error when generating the target URL at the URIBuilder constructor
+	 * @throws ErrorResourceInitialization
+	 *
+	 * @see <a target="_blank" href="https://developers.onelogin.com/api-docs/2/smart-mfa/verify-token">Verify MFA Token documentation</a>
+	 */
+	public boolean verifyToken(String stateToken, String OTPToken) throws OAuthSystemException, OAuthProblemException, URISyntaxException, ErrorResourceInitialization {
+			int versionId = settings.getVersionId("SMART_MFA_VERFY_TOKEN");
+			URIBuilder url = new URIBuilder(settings.getURL(Constants.SMART_MFA_VERFY_TOKEN, versionId));
+
+			Map<String, Object> data = new HashMap<String, Object>();
+			data.put("state_token", stateToken);
+			data.put("otp_token", OTPToken);
+
+			String body = null;
+			if (!data.isEmpty()) {
+				body = JSONUtils.buildJSON(data);
+			}
+
+			OneloginOAuth2JSONResourceResponse oAuth2Response = (OneloginOAuth2JSONResourceResponse) executeCall(url, OAuth.HttpMethod.POST, OneloginOAuth2JSONResourceResponse.class, body);
```

Comment:
From my testing the response from the API is `OK` on a 200 and is not a JSON object... not sure the best way to work with that here.

Reply:
I see there is also a typo on the URL constant
SMART_MFA_VERFY_TOKEN
Should be
SMART_MFA_VERIFY_TOKEN

Reply:
In the code I tried to only check the header, as I expect to get other status code if it fails.

But I agree this is risky. The endpoint should return a valid formated result and not just a OK.

Maybe in order to improve security, avoiding verify_token to retur 
200 on calls to wrong urls that could return a simply 200,
maybe you could as well verify that the body length of the response is short as well as its OK.

Reply:
Python SDK should be improved as well
https://github.com/onelogin/onelogin-python-sdk/blob/v2/src/onelogin/api/client.py#L3222

Reply:
I have a fix in place for dealing with the getting a string of `OK` back, basically a new response type that only parses the body as JSON on an error.